### PR TITLE
feat: improving http parser with simd isa(now support sse4.2 and avx2)

### PR DESCRIPTION
**Pull request by helintongh**
benchmark in 11th Gen Intel(R) Core(TM) i7-11700 @ 2.50GHz:

all use o3 flag.and parse http string 5000000 times.

image

without simd parser http :  Elapsed 4.782222 seconds.
with sse4.2: Elapsed 1.650459 seconds.
with avx2:  Elapsed 3.895595 seconds.

In fact, High-end devices avx2 instruction set will perform better.

### CODE REVIEW

**Comment by helintongh** [comment]
next feature I want to optimize memcpy function by modeled on dpdk.

**Comment by qicosmos** [comment]
Nice, please also add benchmark code.

**Comment by qicosmos** [comment]
I will test it later.

**Comment by helintongh** [comment]
> Nice, please also add benchmark code.

test code and details in there [on github](https://github.com/helintongh/CWheel/tree/master/picohttpparser). I found a weird condition.when header's suffix is hpp benchmark performance will be better.Maybe we should change picohttpparser.h to picohttpparser.hpp.

**Comment by qicosmos** [comment]
Please add note: how to use these macros.

**Comment by helintongh** [comment]
> Please add note: how to use these macros.

add notes.

File: README.md [hunk 1]
```diff
@@ -60,6 +60,17 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fcoroutines")
...
 set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -fno-tree-slp-vectorize")
 
+## cinatra指令集功能使用
+
+cinatra支持通过指令集优化其内部逻辑，其通过宏来控制是否使用指令集。使用之前请确保cpu支持。
+
+使用如下命令即可编译带simd优化的cinatra。
+
+```shell
+cmake -DENABLE_SIMD=SSE42 .. # 启用sse4.2指令集
```
Comment (qicosmos) [inline README.md:70]: 这两个能同时启用吗？

Comment (helintongh) [reply README.md:70]: 不能，文档里再说明一下？

Comment (qicosmos) [reply README.md:70]: 说明一下

Comment (helintongh) [reply README.md:70]: 添加了说明

**Review by qicosmos — APPROVED** [review]
LGTM

**Pull request merged**